File: interval-server/src/main/java/com/interval/timeslot/service/TimeSlotService.java
package com.interval.timeslot.service;

import com.interval.timeslot.dto.DeleteTimeSlotResponseDto;
import com.interval.timeslot.dto.BatchDeleteTimeSlotRequest;
import com.interval.timeslot.dto.BatchDeleteTimeSlotResponseDto;
import com.interval.timeslot.dto.BatchUpsertTimeSlotRequest;
import com.interval.timeslot.dto.BatchUpsertTimeSlotResponseDto;
import com.interval.timeslot.dto.TimeSlotDto;
import com.interval.timeslot.dto.UpsertTimeSlotRequest;

import java.time.LocalDate;
import java.util.List;

public interface TimeSlotService {

    List<TimeSlotDto> getDailySlots(Long userId, LocalDate date);

    TimeSlotDto upsertTimeSlot(Long userId, UpsertTimeSlotRequest request);

    BatchUpsertTimeSlotResponseDto batchUpsertTimeSlots(Long userId, BatchUpsertTimeSlotRequest request);

    DeleteTimeSlotResponseDto deleteTimeSlot(Long userId, Long slotId);

    BatchDeleteTimeSlotResponseDto batchDeleteTimeSlots(Long userId, BatchDeleteTimeSlotRequest request);
}

File: interval-client/src/services/timeSlotService.ts
import { http } from '@/services/http';
import type {
  BatchDeleteTimeSlotRequest,
  BatchDeleteTimeSlotResponseDto,
  BatchUpsertTimeSlotRequest,
  BatchUpsertTimeSlotResponseDto,
  DeleteTimeSlotResponseDto,
  TimeSlotDto,
  UpsertTimeSlotRequest,
} from '@/types/timeSlot';

export async function getDailySlots(date: string): Promise<TimeSlotDto[]> {
  return http.get<never, TimeSlotDto[]>('/api/time-slots', { params: { date } });
}

export async function upsertTimeSlot(payload: UpsertTimeSlotRequest): Promise<TimeSlotDto> {
  return http.put<never, TimeSlotDto>('/api/time-slots', payload);
}

export async function upsertTimeSlotBatch(payload: BatchUpsertTimeSlotRequest): Promise<BatchUpsertTimeSlotResponseDto> {
  return http.post<never, BatchUpsertTimeSlotResponseDto>('/api/time-slots/batch', payload);
}

export async function deleteTimeSlot(slotId: number): Promise<DeleteTimeSlotResponseDto> {
  return http.delete<never, DeleteTimeSlotResponseDto>(`/api/time-slots/${slotId}`);
}

export async function deleteTimeSlotBatch(payload: BatchDeleteTimeSlotRequest): Promise<BatchDeleteTimeSlotResponseDto> {
  return http.delete<never, BatchDeleteTimeSlotResponseDto>('/api/time-slots/batch', { data: payload });
}
